### ISSUE TITLE
Support for java.util.Optional source mappings

### ISSUE THREAD

**Issue by ctrl-alt-dev** [42BV/beanmapper#137]
Optionals are a common usage pattern in JPA Entities to indicate the possibility of a field being null and so replacing null checks with a niceish api.

Currently if an Entity has an `java.util.Optional` getter which needs to be mapped to a non optional Result field BeanMapper:
- in strict mode fails with `Missing matching properties for source [java.util.Optional] > target [EntityResult]`
- in non-strict mode fails by not mapping any fields to the entity.

The following example demonstrates the problem:
(note the mandatory package prefix)
```
package nl.optionalexample;

public class OptionalMappingExample {

    public static class MyEntity {
        private String value = "42";
        private MyEntity child;
        public String getValue() {
            return value;
        }
        public Optional<MyEntity> getChild() {
            return Optional.ofNullable(child);
        }
    }
    
    public static class MyEntityResult {
        public String value;
        public MyEntityResult child;
    }
    
    public static void main(String[] args) {
        BeanMapper mapper = new BeanMapperBuilder()
                .addPackagePrefix("nl")
                .setApplyStrictMappingConvention(true)
                .build();
        mapper.map(new MyEntity(), MyEntityResult.class);
    }
}
```

The error thrown:
```
11:46:54.418 [main] DEBUG io.beanmapper.strategy.MapToClassStrategy -     {
11:46:54.422 [main] ERROR io.beanmapper.core.BeanStrictMappingRequirementsException - Missing matching properties for source [java.util.Optional] > target [nl.videma.repartitie.rightsholder.OptionalMappingExample.MyEntityResult]* for fields:
11:46:54.423 [main] ERROR io.beanmapper.core.BeanStrictMappingRequirementsException - > MyEntityResult.child
11:46:54.423 [main] ERROR io.beanmapper.core.BeanStrictMappingRequirementsException - > MyEntityResult.value
Exception in thread "main" io.beanmapper.core.BeanStrictMappingRequirementsException
	at io.beanmapper.core.BeanMatch.validateBeanMatchValidity(BeanMatch.java:75)
	at io.beanmapper.core.BeanMatch.validateMappingRequirements(BeanMatch.java:68)
```

**Comment by marcus-talbot42** [comment]
An implementation has been proposed in #147

**Comment by marcus-talbot42** [comment]
Implemented by #147
